packages/appium/lib/appium.js

```diff
@@ -516,6 +524,17 @@ class AppiumDriver extends BaseDriver {
     // handling the command and which didn't
     plugins.length && this.logPluginHandlerReport({cmd, cmdHandledBy});
 
+    // And finally, if the command was createSession, we want to migrate any plugins which were
+    // previously sessionless to use the new sessionId, so that plugins can share state between
+    // their createSession method and other instance methods
+    if (cmd === 'createSession' && this.sessionlessPlugins.length && !res.error) {
+      const sessionId = res.value[0];
+      log.info(`Promoting ${this.sessionlessPlugins.length} sessionless plugins to be attached ` +
```

Comment:
btw, if you hate having to split template strings, there's a [lib](https://npm.im/common-tags) containing tagged template functions which make this less ornery

Reply:
yes, i like that! i still find js tagged template functions kind of ugly, and don't want to add an extra lib in this PR but could be a good thing to consider making standard or optional in the future.